src/documents/views.py

```diff
@@ -295,6 +294,12 @@ def original_requested(request):
 
     def file_response(self, pk, request, disposition):
         doc = Document.objects.get(id=pk)
+        if request.user is not None and not has_perms_owner_aware(
```

Comment:
The logic all seems fine, but it's duplicated in enough places I think it should be factored out.  Some sort of method, maybe returning None or HttpResponseForbidden?

Reply:
Yea I thought would be a bit odd (having to e.g. return None if perms are ok) to do that even though it's repeated a bit? I did refactor the meat of this function, as you noticed ofc